cibuildwheel/util.py

```diff
@@ -121,7 +123,42 @@ def __repr__(self):
         return f'{self.__class__.__name__}{self.base_file_path!r})'
 
 
+class Architecture(str, Enum):
+    # mac/linux archs
+    x86_64 = 'x86_64'
+    i686 = 'i686'
+    aarch64 = 'aarch64'
+    ppc64le = 'ppc64le'
+    s390x = 's390x'
+
+    # windows archs
+    x86 = 'x86'
+    AMD64 = 'AMD64'
+
+    @staticmethod
+    def parse_config(config: str, platform: str) -> 'List[Architecture]':
+        result = []
+        for arch_str in re.split(r'[\s,]+', config):
+            if arch_str == 'auto':
+                result += Architecture.auto_archs(platform=platform)
+            else:
+                result.append(Architecture(arch_str))
+        return result
+
+    @staticmethod
+    def auto_archs(platform: str) -> 'List[Architecture]':
```

Comment:
Same here about `native`? The variable names inside are `native`, so why the function not as well?
(depends on `--arch auto` vs. `--arch native`, though)

Reply:
This part of the code reflects what I commented earlier `auto` and `native` are not the same. It's very clear with this function.

Reply:
i'm happy with either, but I agree with @mayeut's argument that `i686` isn't 'native' on an x86_64 machine, so `auto` makes more sense to me